spec/controllers/api/mentee_applications_controller_spec.rb

```diff
@@ -1,27 +1,83 @@
-require "spec_helper"
+require 'rails_helper'
 
-describe Api::MenteeApplicationsController, type: :controller do
-  describe "POST /api/mentee_applications" do
-    before do
-      ActionMailer::Base.deliveries.clear
-    end
+RSpec.describe Api::MenteeApplicationsController, type: :controller do
+	describe "Api Mentee Application controller tests" do
+		before(:each) do
+			@edition = create(:edition)
+		end
 
-    it "should create new mentee application for proper params" do
-      applications_number = MenteeApplication.count
+		it 'should start create a Mentee Application, step 1' do
+			post :create, application: {
+				first_name: "Mentee", last_name: "Rspec", email: "mentee@email.com",
+				gender: "female", country: "IN", program_country: "IN",
+				time_zone: "5 - Mumbai", communicating_in_english: "true",
+				send_to_mentor_confirmed: "true",
+				time_availability: 3,
+				previous_programming_experience: "false" },
+				step: "1", steps: "3"
 
-      expect(ActionMailer::Base.deliveries.count).to eq(0)
+			expect(response).to have_http_status(200)
+		end
 
-      params = Fabricate.to_params(:mentee_application, programming_language: ProgrammingLanguage.pluck(:slug).sample)
-      params = params.each{|k, v| params[k] = v.to_s}
+		it 'should continue to create a Mentee Application, step 2' do
+			post :create, application: {
+				first_name: "Mentee", last_name: "Rspec", email: "mentee@email.com",
```

Comment:
you can declare those params in eg separate method as same code is being repeated in all 4 tests

Reply:
I fixed that, I put all common params in a hash